include/crm/common/logging.h

```diff
@@ -134,6 +134,31 @@ void pcmk_log_xml_impl(uint8_t level, const char *text, const xmlNode *xml);
 #    define CRM_TRACE_INIT_DATA(name) QB_LOG_INIT_DATA(name)
 #endif
 
+/*!
+ * \internal
+ * \brief Clip log_level to \p uint8_t range
+ * \deprecated Do not Pacemaker for general-purpose logging
+ *
+ * \param[in] level  Log level to clip
+ *
+ * \return 0 if \p level is less than 0, 255 if \p level is greater than 255, or
+ *         \p level otherwise
+ */
+/* @COMPAT: Make this function internal at a compatibility break. It's used in
+ * public macros for now.
+ */
+static inline uint8_t
+pcmk_clip_log_level(int level)
+{
+    if (level <= 0) {
+        return 0;
+    }
+    if (level >= 255) {
+        return 255;
```

Comment:
This might be too paranoid, but you can use UINT8_MAX out of stdint.h instead of 255.

Reply:
Good call. I prefer that not for paranoia but for clarity. I didn't find a constant in limits.h. It slipped my mind that it would be in stdint.h with the type definition.